Tue Dec  7 16:45:10 EST 2010  "Roland Haas" <roland.haas@physics.gatech.edu>
  * visitCarpetHDF5: add fudge env variable to work around white corners issue
  
    if refined domain do not end on a coarse grid points then VisIt (since it is
    cell based) will draw a white stripe on one side of the domain. This restores
    Christians old code which accidentally fills in data for 2d data
    Starting VisIt as "VISIT_2D_GRANULARITY_FUDGE=1 visit" activates the fudge.
diff -rN -u old-tools/visitCarpetHDF5/avtCarpetHDF5FileFormat.C new-tools/visitCarpetHDF5/avtCarpetHDF5FileFormat.C
--- old-tools/visitCarpetHDF5/avtCarpetHDF5FileFormat.C	2011-07-12 09:18:51.000000000 -0400
+++ new-tools/visitCarpetHDF5/avtCarpetHDF5FileFormat.C	2011-07-12 09:18:51.000000000 -0400
@@ -266,6 +266,7 @@
     int num_dims = dim;   // defined in vtCarpetHDF5FileFormat.h as 'static const int dim = 3;'
     int num_levels = data_file->timesteps[timeState].max_cart_rl;
     int num_patches = data_file->timesteps[timeState].cart_comp[0].size();
+    int vanilla_code = getenv("VISIT_2D_GRANULARITY_FUDGE") ? atoi(getenv("VISIT_2D_GRANULARITY_FUDGE"))!=0 : 0;
 
     // first, look to see if we don't already have it cached
     void_ref_ptr vrTmp = cache->GetVoidRef("any_mesh",
@@ -323,14 +324,14 @@
                 float c0 = comp[0][j].zmin();
                 float c1 = comp[0][j].zmax();
 
-                float dx = comp[0][j].delta()[0]/2;
-                float dy = comp[0][j].delta()[1]/2;
-                float dz = comp[0][j].delta()[2]/2;
+                float dx = !vanilla_code * comp[0][j].delta()[0]/2;
+                float dy = !vanilla_code * comp[0][j].delta()[1]/2;
+                float dz = !vanilla_code * comp[0][j].delta()[2]/2;
 
                 // compare with a half-finer level offset to work around floating point roundoff issues
-                if (((dims[0] > 1) && (a0 >= x1-dx || x0 >= a1-dx)) ||
-                    ((dims[1] > 1) && (b0 >= y1-dy || y0 >= b1-dy)) ||
-                    ((dims[2] > 1) && (c0 >= z1-dz || z0 >= c1-dz)))
+                if (((dims[0] > 1 || vanilla_code) && (a0 >= x1-dx || x0 >= a1-dx)) ||
+                    ((dims[1] > 1 || vanilla_code) && (b0 >= y1-dy || y0 >= b1-dy)) ||
+                    ((dims[2] > 1 || vanilla_code) && (c0 >= z1-dz || z0 >= c1-dz)))
                     continue;
 
                 childPatches.push_back(j);
